Automattic/wp-calypso
# Updates the promote post behavior for sync in progress error

When you first connect Jetpack to WPCOM, there is a delay for the post to get synced. During that time, if you access the Blaze Dashboard, you will see a blocking error preventing you from creating a campaign until the site finishes sinking. 

The main problem is that the user can't promote old posts because we won't have information about them in WPCOM, but new posts will be ready to promote as soon as they are created. 

To improve this error, we are changing the message to be a warning and not blocking the result of the `/blaze/posts` endpoint.

## Proposed Changes

* Adds a behavior to handle warning messages from the `/blaze/posts` endpoint

## Why are these changes being made?

* We want to improve the first contact of the users with the dashboard, when they are using an external site with Jetapck or any other Jetpack-connected plugin.

## Testing Instructions

Jetpack still sends an error on that endpoint, so the only way to test this before we merged a change in Jetpack is by hardcoding the response of the endpoint (or changing it on a sandbox).

I will mention the step for the hardcode approach, but feel free to sandbox the public-api and change the `/blaze/posts` endpoint to start returning the new `warnings` prop.

* Checkout this branch and start the Server
* Edit this file and change the line 91 to:
```
warnings: [ 'sync_in_progress' ],
```
* Navigate to Tools->Advertising
* Verify that you see the new Info message and the full list of posts.

![Screenshot 2024-09-24 at 5 03 04 PM](https://github.com/user-attachments/assets/aed5e73c-08a9-4fe4-8089-2c06d3bdecea)

## Pre-merge Checklist
